lib/Github/AuthMethod.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Github;
+
+final class AuthMethod
+{
+    /**
+     * Authenticate using a client_id/client_secret combination.
+     *
+     * @var string
+     */
+    const AUTH_CLIENT_ID = 'client_id_header';
```

Comment:
I woud remove the `AUTH_` prefix for these constants as the classname gives that context already (same for the other constants)